client/app/pages/dashboards/hooks/useDashboard.js

```diff
@@ -157,6 +157,13 @@ function useDashboard(dashboardData) {
     );
   }, [dashboard]); // eslint-disable-line react-hooks/exhaustive-deps
 
+  const copyDashboard = useCallback(() => {
+    const dashboardSlug = dashboard.slug;
+    Dashboard.copy({ slug: dashboardSlug }).then(({ slug }) => {
+      window.open(`dashboard/${slug}`);
+    });
+  }, [dashboard]);
```

Comment:
Please refer to the similar function we have for queries. It's implemented in a bit cumbersome way not because we were high, but because that was the only way to make it reliable. E.g. if you'll try to open a popup in some async handler (like here) - 50/50 browser will block it, because JS can show popups only as a response to user interaction. Also it's important to disable menu item while action is in progress. And so on.

Reply:
@yankeeguyu Please address this one too. Use this for the reference: https://github.com/getredash/redash/blob/83726da48a0d1768a8f726f007ce0c6fd594c2de/client/app/pages/queries/hooks/useDuplicateQuery.js

Reply:
Sorry, i am going to modify this later!

Reply:
Hi @kravets-levko , recently i have no time to modify this comment, perhaps you can modify this comment! Sorry about that!

Reply:
> @yankeeguyu Please address this one too. Use this for the reference: https://github.com/getredash/redash/blob/83726da48a0d1768a8f726f007ce0c6fd594c2de/client/app/pages/queries/hooks/useDuplicateQuery.js

Hi @kravets-levko , i modified this comment, but it can't work! So someone can help me? Thank you!